suite-common/wallet-core/src/fiat-rates/fiatRatesSelectors.ts

```diff
@@ -0,0 +1,211 @@
+import { A, D, F, pipe } from '@mobily/ts-belt';
+import { memoize, memoizeWithArgs } from 'proxy-memoize';
+
+import { FiatCurrencyCode } from '@suite-common/suite-config';
+import {
+    Account,
+    AccountKey,
+    WalletAccountTransaction,
+    FiatRateKey,
+    FiatRatesStateLegacy,
+    Rate,
+    RateType,
+    TickerId,
+} from '@suite-common/wallet-types';
+import { getFiatRateKeyFromTicker } from '@suite-common/wallet-utils';
+
+import {
+    AccountsRootState,
+    selectAccountByKey,
+    selectDeviceAccounts,
+} from '../accounts/accountsReducer';
+import { TransactionsRootState, selectTransactions } from '../transactions/transactionsReducer';
+import { MAX_AGE, ONE_WEEK_IN_MS } from './fiatRatesConstants';
+import { FiatRatesRootState } from './fiatRatesTypes';
+
+export const selectFiatRatesByFiatRateKey = (
+    state: FiatRatesRootState,
+    fiatRateKey: FiatRateKey,
+    rateType: RateType = 'current',
+): Rate | undefined => state.wallet.fiat?.[rateType]?.[fiatRateKey];
+
+export const selectIsFiatRateLoading = (
+    state: FiatRatesRootState,
+    fiatRateKey: FiatRateKey,
+    rateType: RateType = 'current',
+) => {
+    const currentRate = selectFiatRatesByFiatRateKey(state, fiatRateKey, rateType);
+    return currentRate?.isLoading ?? false;
+};
+
+export const selectIsTickerLoading = (
+    state: FiatRatesRootState,
+    ticker: TickerId,
+    fiatCurrency: FiatCurrencyCode,
+    rateType: RateType = 'current',
+) => {
+    const fiatRateKey = getFiatRateKeyFromTicker(ticker, fiatCurrency);
+    return selectIsFiatRateLoading(state, fiatRateKey, rateType);
+};
+
+export const selectShouldUpdateFiatRate = (
+    state: FiatRatesRootState,
+    fiatRateKey: FiatRateKey,
+    rateType: RateType = 'current',
+) => {
+    const currentRate = selectFiatRatesByFiatRateKey(state, fiatRateKey, rateType);
+
+    if (!currentRate) {
+        return true;
+    }
+
+    const { lastSuccessfulFetchTimestamp } = currentRate;
+    const now = Date.now();
+
+    return now - lastSuccessfulFetchTimestamp > MAX_AGE[rateType];
+};
+
+export const selectTickerFromAccounts = memoize((state: FiatRatesRootState): TickerId[] => {
+    const accounts = selectDeviceAccounts(state as any);
+    return pipe(
+        accounts,
+        A.map(account => [
+            {
+                symbol: account.symbol,
+            } as TickerId,
+            ...(account.tokens || []).map(
+                token =>
+                    ({
+                        symbol: account.symbol,
+                        tokenAddress: token.contract,
+                    }) as TickerId,
+            ),
+        ]),
+        A.flat,
+        F.toMutable,
+    );
+});
+
+export const selectTickersToBeUpdated = memoizeWithArgs(
```

Comment:
It does not update rates when using `memoizeWithArgs`, it starts to return `[]`.

cc @matejkriz Should we use it on methods in this file?

Reply:
It is called once per 2 minutes. I think it is not required to memoize it. 

Btw, I am not 100% sure I successfully simulated conditions for `selectTransactionsWithMissingRates`. However, I think there is a similar problem

Reply:
I don't think we need memoization here for desktop, not sure about mobile.

Reply:
Removed